### PR TITLE
Change repo structure, add workflow for scattering many files and add cwltests

### PR DESCRIPTION
**Pull request by Vkale1**

### CODE REVIEW

File: src/workflows/runCRISPRCasFinderl.cwl [hunk 1]
```diff
@@ -26,11 +26,10 @@ inputs:
   spacer_fasta:
     type: string?
     label: Fasta with spacer sequences
-    default: "CRISPRCasFinder_Spacers.fasta"
   crisprcas_gff:
     type: string?
     label: GFF3 with CRISPRCas results
-    default: "CRISPRCasFinder.gff3"
```
Comment (caballero) [inline src/workflows/runCRISPRCasFinderl.cwl:33]: if there are no defaults, the value should be enforced by removing "?" from the type definition

File: src/workflows/runCRISPRCasFinderl.cwl [hunk 2]
```diff
@@ -26,11 +26,10 @@ inputs:
   spacer_fasta:
     type: string?
     label: Fasta with spacer sequences
-    default: "CRISPRCasFinder_Spacers.fasta"
```
Comment (caballero) [inline src/workflows/runCRISPRCasFinderl.cwl:29]: if there are no defaults, the value should be enforced by removing "?" from the type definition

**Review by caballero — CHANGES_REQUESTED** [review]
minor comments on removing defaults, almost all changes are fine

File: src/workflows/runCRISPRCasFinderl.cwl (hunk 2 above)
Comment (Vkale1) [reply src/workflows/runCRISPRCasFinderl.cwl:29]: Removed input for both in new commit as it's not required anymore. These string defaults are now defined in the separate tool cwls.

File: tests/tests.yml [hunk 3]
```diff
@@ -6,11 +6,11 @@
   output:
     crisprcasfinder_json:
       basename: result.json
-      #checksum: sha1$36317ca0daa17ba411a75e40ddc44dc86fdf75ff
+      #no checksum as file has date run at the start
```
Comment (mberacochea) [inline tests/tests.yml]: Use `Any` instead

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: Testing
```
Comment (mberacochea) [inline .github/workflows/test.yml:1]: Very nice :1st_place_medal:

File: src/workflows/runCRISPRCasFinderl_array.cwl [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  sequences:
+    type: File[]
+    format: edam:format_1929
+    label: input Fasta file to analyze
+  soFile:
+    type: string?
+    label: path to the sel392v2.so file, required by vmatch
+    default: /opt/CRISPRCasFinder/sel392v2.so
+  casDefinition:
+    type: string?
```
Comment (mberacochea) [inline src/workflows/runCRISPRCasFinderl_array.cwl]: This one could be an Enum instead -> https://www.commonwl.org/user_guide/misc/#enum-inputs

File: src/workflows/runCRISPRCasFinderl_array.cwl [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+steps:
+    full_wf:
+      label: run full CRIPRCasFinder workflow for multiple files
+      run: runCRISPRCasFinderl.cwl
+      scatter: sequences
+      in:
+        sequences: sequences
+        soFile: soFile
+        casDefinition: casDefinition
```
Comment (mberacochea) [inline src/workflows/runCRISPRCasFinderl_array.cwl]: We should try to keep the naming consistent:
`casDefinition` and then the outputs are `crisprcasfinder_fasta`
I would go with `cas_definition`

**Review by mberacochea — APPROVED** [review]
Great stuff! I've added some minor comments, but it looks great :).

Thank you for making the repo "compliant" with the team conventions.

**Pull request merged**